cuqi/implicitprior/_denoiser.py

```diff
@@ -0,0 +1,108 @@
+#%%
+from abc import ABC, abstractmethod
+from cuqi.distribution import Distribution
+import numpy as np
+
+class DenoiseRegularizer(Distribution):
+    """    
```

Comment:
Also, on the name "regularizer", the class is subclassed from distribution and as such logically one would expect to think of it as a (perhaps implicit, not fully specified) distribution or prior, rather than a traditional regularizer, in the sense that roughly distribution = exp(-regularizer).  Wouldn't it be more accurate that its name reflects that it is in the type distribution / implicit prior?  Or have I misunderstood it?

Reply:
Do you think ImplicitDenoisingPrior would be better?
Maybe we could meet to talk about it.